### ISSUE TITLE
404 error using API

### ISSUE THREAD

**Issue by glederrey** [nismod/microsimulation#25]
Dear authors, 

I'm trying to run your code to generate a synthetic population based on the UK census. However, I have the following issue when running the example:

```
---------------------------------------------------------------------------
ParserError                               Traceback (most recent call last)
<ipython-input-5-8e7acef0d92d> in <module>
      6 
      7         # init microsynthesis
----> 8         ssm = Static.SequentialMicrosynthesis(region, resolution, variant, is_custom, cache_dir, output_dir, use_fast_mode)
      9         ssm.run(ref_year, horizon_year)
     10 

D:\Users\glede\anaconda3\envs\spenser\lib\site-packages\microsimulation-0.0.1-py3.6.egg\microsimulation\static.py in __init__(self, region, resolution, variant, is_custom, cache_dir, output_dir, fast_mode)
     24   def __init__(self, region, resolution, variant, is_custom=False, cache_dir="./cache", output_dir="./data", fast_mode=False):
     25 
---> 26     common.Base.__init__(self, region, resolution, cache_dir)
     27 
     28     self.output_dir = output_dir

D:\Users\glede\anaconda3\envs\spenser\lib\site-packages\microsimulation-0.0.1-py3.6.egg\microsimulation\common.py in __init__(self, region, resolution, cache_dir)
     20     self.resolution = resolution
     21     self.data_api_en = Api_ew.Nomisweb(cache_dir)
---> 22     self.data_api_sc = Api_sc.NRScotland(cache_dir)
     23 
     24   def get_census_data(self):

~\AppData\Roaming\Python\Python36\site-packages\ukcensusapi\NRScotland.py in __init__(self, cache_dir)
     83           fd.write(chunk)
     84 
---> 85     self.area_lookup = pd.read_csv(str(self.cache_dir / "sc_lookup.csv"))
     86 
     87     # TODO use a map (just in case col order changes)

~\AppData\Roaming\Python\Python36\site-packages\pandas\io\parsers.py in read_csv(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)
    686     )
    687 
--> 688     return _read(filepath_or_buffer, kwds)
    689 
    690 

~\AppData\Roaming\Python\Python36\site-packages\pandas\io\parsers.py in _read(filepath_or_buffer, kwds)
    458 
    459     try:
--> 460         data = parser.read(nrows)
    461     finally:
    462         parser.close()

~\AppData\Roaming\Python\Python36\site-packages\pandas\io\parsers.py in read(self, nrows)
   1196     def read(self, nrows=None):
   1197         nrows = _validate_integer("nrows", nrows)
-> 1198         ret = self._engine.read(nrows)
   1199 
   1200         # May alter columns / col_dict

~\AppData\Roaming\Python\Python36\site-packages\pandas\io\parsers.py in read(self, nrows)
   2155     def read(self, nrows=None):
   2156         try:
-> 2157             data = self._reader.read(nrows)
   2158         except StopIteration:
   2159             if self._first_chunk:

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader.read()

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader._read_low_memory()

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader._read_rows()

pandas\_libs\parsers.pyx in pandas._libs.parsers.TextReader._tokenize_rows()

pandas\_libs\parsers.pyx in pandas._libs.parsers.raise_parser_error()

ParserError: Error tokenizing data. C error: Expected 1 fields in line 11, saw 4
```

I had a look at the CSV file that is produced while initializing the class `Api_sc.NRScotland(cache_dir)`. It is instead returning an HTML file with a 404 error inside. 

Would you be kind enough to help me fix this issue with the API?

Thank you in advance

**Comment by tomalrussell** [comment]
Hi @glederrey thanks for the error report.

This is bubbling up from our related package, [UKCensusAPI](https://github.com/virgesmith/UKCensusAPI/) and I think this particular issue has been fixed in version 1.1.6, which is unreleased but available if you pull the source from GitHub, master branch. I came across a related connection issue in a second helper package, so you might also need latest master of [ukpopulation](https://github.com/nismod/ukpopulation).

Here's the example I ran to reproduce:

```json5
// example.json
{
  "resolution": "MSOA11",
  "projection": "ppp",
  "census_ref_year": 2011,
  "horizon_year": 2039,
  "mode": "fast",
  "cache_dir": "./cache",
  "output_dir": "./data"
}
```

```bash
python scripts/run_ssm.py -c example.json  E07000178
```

Are you able to pull and install from source to see if this lets you make progress?
